src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java

```diff
@@ -920,4 +921,164 @@ public static void createSourceBQUpdateTable() throws IOException, InterruptedEx
+
+    bqSourceTable = "E2E_TARGET_" + UUID.randomUUID().toString().replaceAll("-", "_");
+    PluginPropertyUtils.addPluginProp("bqSourceTable", bqSourceTable);
+    BeforeActions.scenario.write("BQ source table name - " + bqSourceTable);
+
+
+    io.cdap.e2e.utils.BigQueryClient.getSoleQueryResult("create table `" + datasetName + "." + bqSourceTable + "` " +
+            "(string_value STRING, int_value INT64, float_value FLOAT64,boolean_value BOOL)");
+
+
+
+    try {
+      io.cdap.e2e.utils.BigQueryClient.getSoleQueryResult("INSERT INTO `" + datasetName + "." + bqSourceTable + "` " +
+              "(string_value,  int_value, float_value,boolean_value)" +
+              "VALUES" +  "('string_1', 1, 0.1,true)," +
+              "('string_1', 2, 0.2,false)");
+
+
+
+    } catch (NoSuchElementException e) {
+      // Insert query does not return any record.
+      // Iterator on TableResult values in getSoleQueryResult method throws NoSuchElementException
+    }
+
+    PluginPropertyUtils.addPluginProp(" bqSourceTable",  bqSourceTable);
+    BeforeActions.scenario.write("BQ Source Table " +  bqSourceTable + " updated successfully");
+  }
+
+
+  @Before(value = "@BQ_NULL_MODE_SOURCE_TEST")
+  public static void createNullSourceBQTable() throws IOException, InterruptedException {
+    bqSourceTable = "E2E_TARGET_" + UUID.randomUUID().toString().replaceAll("-", "_");
+    PluginPropertyUtils.addPluginProp("bqSourceTable", bqSourceTable);
+    BeforeActions.scenario.write("BQ source table name - " + bqSourceTable);
+    io.cdap.e2e.utils.BigQueryClient.getSoleQueryResult("create table `" + datasetName + "." + bqSourceTable + "` " +
+            "(Address STRING, id INT64, Firstname STRING,LastName STRING)");
+
+    try {
+      io.cdap.e2e.utils.BigQueryClient.getSoleQueryResult("INSERT INTO `" + datasetName + "." + bqSourceTable + "` " +
+              "(Address,  id, Firstname, LastName)" +
+              "VALUES" +  "('Agra', 1, 'Neeraj','')," +
+              "('Noida', 2, '','')");
+
+    } catch (NoSuchElementException e) {
+      // Insert query does not return any record.
+      // Iterator on TableResult values in getSoleQueryResult method throws NoSuchElementException
+    }
+
+    PluginPropertyUtils.addPluginProp(" bqSourceTable",  bqSourceTable);
+    BeforeActions.scenario.write("BQ Source Table " +  bqSourceTable + " updated successfully");
+  }
+
+
+
+
+
+
+  @Before(value = "@BQ_TIME_STAMP_SOURCE_TEST")
+  public static void createTimeStampBQTable() throws IOException, InterruptedException {
+    bqSourceTable = "E2E_TARGET_" + UUID.randomUUID().toString().replaceAll("-", "_");
+    PluginPropertyUtils.addPluginProp("bqSourceTable", bqSourceTable);
+    BeforeActions.scenario.write("BQ source table name - " + bqSourceTable);
+    BigQueryClient.getSoleQueryResult("create table `" + datasetName + "." + bqSourceTable + "` " +
+            "(Address STRING, transaction_date DATE, Firstname STRING)");
```

Comment:
As per ITN class data should be like: *  {"string_value":"string_1","int_value":1,"float_value":0.1,"date_value":"2000-01-19"}. You can change the data values but, fields should be there of type string,int,float and date